src/goose.rs

```diff
@@ -698,24 +707,46 @@ impl GooseClient {
         load_test_hash: u64,
     ) -> Self {
         trace!("new client");
-        GooseClient {
-            task_sets_index,
-            base_url: Arc::new(RwLock::new(base_url)),
-            min_wait,
-            max_wait,
-            config: configuration.clone(),
-            parent: None,
-            // A value of max_value() indicates this client isn't fully initialized yet.
-            weighted_clients_index: usize::max_value(),
-            weighted_on_start_tasks: Vec::new(),
-            weighted_tasks: Vec::new(),
-            weighted_on_stop_tasks: Vec::new(),
-            task_request_name: None,
-            request_name: None,
-            load_test_hash,
+        match Client::builder()
+            .user_agent(APP_USER_AGENT)
+            .cookie_store(true)
+            .build()
+        {
+            Ok(c) => {
+                GooseClient {
+                    task_sets_index,
+                    client: Arc::new(Mutex::new(c)),
```

Comment:
Unrelated to that change - I recently learned that boxing things can make structures smaller and that can under certain circumstances avoid memcpy calls (if the whole structure < 128 bytes). Not sure this applies here though, but it came to mind as reqwest is pretty large - on the other hand there is one more indirection age when accessing the inner structures.

In how rust works it’s even feasible to just create one boxed inner struct and then refer to it like client.0.original_property.

Really just food for thought - the changes here are great.